src/app/api/services/content-types-info/content-types-info.service.ts

```diff
@@ -77,19 +77,36 @@ export class ContentTypesInfoService {
     }
 
     private getItem(type: string, prop: string): string {
-        // Some endpoints have different name/labels for the same content types
+        let result: any;
+
+        if (type) {
+            type = this.getTypeName(type);
+
+            for (let i = 0; i < this.contentTypeInfoCollection.length; i++) {
+                let item = this.contentTypeInfoCollection[i];
+                if (item.clazz.toLocaleLowerCase() === type.toLocaleLowerCase() || item.label.toLocaleLowerCase() === type.toLocaleLowerCase()) {
+                    result = item[prop];
+                    break;
+                }
+            }
+        }
+
+        return result;
+    }
+
+    /**
+     * Some endpoints have different name/labels for the same content types
+     * @param type
+     */
+    private getTypeName(type: string): string {
         if (type === 'HTMLPAGE' || type === 'htmlpage') {
```

Comment:
This might be in a Map such as 
type -> typeName

so you only have to do something like

return  typeNameMap[type];

will be log(N) instead of O(N) since won't be iterative testing each case.